apps/l10nstats/management/commands/progress.py

```diff
@@ -121,6 +124,25 @@ def rescale(self, vals):
         scale = lambda v: (v - _min) * ratio + bottom
         return [(t, scale(c)) for t, c in vals]
 
+    def initialCoverage(self, tuples, startdate):
+        rv = {}
+        tree2locs = defaultdict(list)
+        for l,t in tuples:
+            tree2locs[t].append(l)
+        for tree, locales in tree2locs.iteritems():
+            locs = (Locale.objects
+                .filter(code__in=locales)
+                .filter(run__srctime__lt=startdate, run__tree__code=tree)
```

Comment:
It's on Run, https://github.com/mozilla/elmo/blob/master/apps/l10nstats/models.py#L39, that has foreign keys to both Locale and Tree (both in life/models.py)